### ISSUE TITLE
What JRE to use with Jython now?

### ISSUE THREAD

**Issue by Mrodent** [jython/jython#83]
This may be a naive question. 

I have a couple of Jython projects which I'm just revising a bit ... previously they linked to the .jars from Java 8 JRE. But now my system uses Java 11.  There is no Oracle JRE for Java 11, and even if you use an OpenJDK-based JDK the JRE doesn't seem to be delivered with .jars.

Maybe it is a bit incongruous to be using a "Python 2.7" with Java 11? Nevertheless, this is the Java version I use on my system. In the Python world, we have virtual environments, and there is such a thing as "sdkman" in Linux. This particular machine of mine is running W10 however.  Since Java 9 we also have these annoying 6 warnings whenever we run a Jython project ("illegal reflective access blah blah").

BTW, bring on Jython3 guys! ... I just read issue #24 from a year ago, last comment last August.  That would be so great, and I am glad to see that Jeff5 is still keen.

At the moment I am using the .jars from Oracle JRE 10 even though my system is 11.  Is this recommended or a bad idea?  Apparently you are meant to be able to "build" your own JRE with v11. But given that I got hold of AdoptOpenJDK's JRE, and could see no .jars, that would probably be wasted effort. 

In short, is Jython currently capable of using Java v11's classes ecosystem?

**Comment by jeff5** [comment]
I have a newbie-written PowerShell script for switching versions of Java, Python and Jython at the prompt, as I often have multiple such windows up at the same time. It just edits the path.

I don't really understand the opening comments about JRE availability. I have a wide range of versions on my (Windows) system, but I always install a JDK. Oracle have changed their license, but what I do seems to be within it.

Jython 2.7.2 works for me on Java 11.
```
PS 272-trial> Set-EnvJython 2
java version "1.8.0_271"Java(TM) SE Runtime Environment (build 1.8.0_271-b09)Java HotSpot(TM) 64-Bit Server VM (build 25.271-b09, mixed mode)
Jython 2.7.2
PS 272-trial> Set-EnvJava 11
java version "11.0.9" 2020-10-20 LTSJava(TM) SE Runtime Environment 18.9 (build 11.0.9+7-LTS)Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.9+7-LTS, mixed mode)
PS 272-trial> jython
Jython 2.7.2 (v2.7.2:925a3cc3b49d, Mar 21 2020, 10:03:58)
[Java HotSpot(TM) 64-Bit Server VM (Oracle Corporation)] on java11.0.9
Type "help", "copyright", "credits" or "license" for more information.
>>>
```
We [test on OpenJDK 8, 11 and 13](https://github.com/jython/jython/blob/master/.travis.yml) thanks largely to a patch by @cclauss .

We did our best to get rid of the reflective access warnings: I think we have from our own code. They are emitted from JARs we depend on, pretty much only when you ask for a C-style file handle I think.  There are options on the `java` command that will reduce (eliminate?) these checks, but it didn't seem right to build that into Jython (say in the launcher).

> Maybe it is a bit incongruous to be using a "Python 2.7" with Java 11? 

It's uncomfortable to be using Python 2.7 at all :-/ but there isn't really a correlation to Java versions. Language concepts that appeared after about Java 7 (e.g. lambdas) are not supported.

**Comment by chris-caat** [comment]
Thanks... dare I ask... any news on the Holy Grail aka Jython 3.x? In the thread I referenced one contributor actually mentioned the possibility of funding.

PS Personally, I'm inured to those silly reflection warnings as I use Groovy quite a bit.

**Comment by jeff5** [comment]
@Mrodent : I see what you mean about he JRE 11 now. Is this SO article helpful: https://stackoverflow.com/questions/53733312/where-is-jre-11

Jython 3: Working on it but off-repo where I can try wild stuff. I want it to be shaped by efficient use of the dynamic language features as there should be a performance advantage from them. Beating Jython 2 is proving difficult, which I take to mean I'm not doing it right yet.
